docs/tutorials/advanced/using_the_cache.rst

```diff
@@ -0,0 +1,99 @@
+.. _using-the-cache:
+
+Using the cache
+===============
+
+Whilst for stochastic strategies every repetition of a Match will give a
+different result. For deterministic strategies, when there is no noise there is
+no need to re run the match. The library has a :code:`DeterministicCache` class
+that allows us to quickly replay matches.
+
+Here is how to create a new empty cache::
+
+    >>> import axelrod as axl
+    >>> cache = axl.DeterministicCache()
+    >>> len(cache)
+    0
+
+Caching a Match
+---------------
+
+We can use this cache in a Match, let us also time the execution::
+
+    >>> import timeit
+    >>> def run_match():
+    ...     p1, p2 = axl.GoByMajority(), axl.Alternator()
+    ...     match = axl.Match((p1, p2), deterministic_cache=cache, turns=200)
+    ...     return match.play()
+    >>> time_with_empty_cache = timeit.timeit(run_match, number=500)
+
+We can take a look at the cache::
+
+    >>> cache  # doctest: +ELLIPSIS
+    {(<class 'axelrod.strategies.gobymajority.GoByMajority'>, <class 'axelrod.strategies.alternator.Alternator'>, 200): [('C', 'C'), ..., ('C', 'D')]}
+    >>> len(cache)
+    1
+
+Note, that is maps triplets of players and match lengths to interactions.
```

Comment:
Might be better as:

Note: This maps a triplet of 2 player classes and the match length to the resulting interactions.